src/components/d3/gauge-arc.js

```diff
@@ -0,0 +1,104 @@
+import PropTypes from 'prop-types'
+import React, { useRef, useEffect, useMemo } from 'react'
+import * as d3 from 'd3'
+
+const GaugeArc = ({ data, width, height, config }) => {
+
+  const { color, backgroundColor, dataKey, transition } = config
+
+  const value = data[dataKey.x2] ? data[dataKey.x1] / data[dataKey.x2] * 100 : 0
+
+  const remainder = 100 - value
+  const colors = [color, backgroundColor]
+  const svgRef = useRef(null)
+
+  const isAscending = value > remainder
+
+  const pieSize = width > height ? height / 2 : width / 2
+
+  const lineThickness = pieSize * 0.2
+
+  const arcsGenerator = d3.pie().sortValues(function (a, b) { return isAscending ? b - a : a - b })
+
+  const arcsData = useMemo(() => arcsGenerator([value, remainder]), [data])
```

Comment:
I think here you are safe replacing data with'arcsGenerator', 'remainder', and 'value'